src/span.h

```diff
@@ -290,6 +292,15 @@ inline const unsigned char* UCharCast(const std::byte* c) { return reinterpret_c
 template <typename B>
 concept BasicByte = requires { UCharCast(std::span<B>{}.data()); };
 
+inline constexpr std::span<const uint8_t> BytesToUInt8s(const std::span<const std::byte> s) noexcept
+{
+    return std::bit_cast<std::span<const uint8_t>>(s);
```

Comment:
Based on what I read C++23 solves some of these problems (e.g. `Every specialization of std::span is a TriviallyCopyable type. (since C++23)`) and while array and vector casts would likely be safe, thanks for pointing out that the layout may not always be the same and we need something more predictable here.
Wanted to avoid explicit pointer arithmetic or extra copying, let me know if the latest push is a good compromise in this area.